### PR TITLE
Ubdate advanced search

### PR DESCRIPTION
**Pull request by lyzunyktaras**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

Update advanced search

## Summary of change

ToDo

Add name to advanced serach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Comment by lyzunyktaras** [comment]
These changes allow us to use the name field in the advanced search to search clubs with filters including club name. I added the name field to AdvancedSearchClubProfile and rewrote the query for the correct search.

File: src/main/java/com/softserve/teachua/dto/search/AdvancedSearchClubProfile.java [hunk 1]
```diff
@@ -9,6 +9,8 @@
 @AllArgsConstructor
 @Data
 public class AdvancedSearchClubProfile {
+    private String name;
```
Comment (marianben) [inline src/main/java/com/softserve/teachua/dto/search/AdvancedSearchClubProfile.java:12]: Why didn't you add an annotation  @Nullable to the field "name"?

File: src/main/java/com/softserve/teachua/repository/ClubRepository.java [hunk 2]
```diff
@@ -47,8 +47,12 @@
             + "(:age IS NULL or club.ageFrom <= :age AND club.ageTo >= :age) AND "
             + "((:categories) IS NULL OR category.name IN (:categories)) AND "
             + "(:district IS NULL OR district.name = :district) AND "
+            + "(:name IS NULL OR "
+            + "LOWER(club.name) LIKE LOWER(CONCAT('%', :name , '%')) OR "
+            + "LOWER(club.description) LIKE LOWER(CONCAT('%', :name , '%'))) AND "
```
Comment (marianben) [inline src/main/java/com/softserve/teachua/repository/ClubRepository.java:52]: Can you attach search results? (I want to see the search result screen only with name and without the name)

File: src/test/java/com/softserve/teachua/repository/ClubRepositoryIT.java [hunk 3]
```diff
@@ -399,6 +414,7 @@ public void advancedSearchWithAllCorrectParamsShouldReturnEmptyList() {
         ClubSearchTestEntity testClub = ClubAdvancedSearchRepository.getClubAsListWithAllCorrectData();
 
         List<Club> clubs = clubRepository.findAllBylAdvancedSearch(
+                null,
```
Comment (marianben) [inline src/test/java/com/softserve/teachua/repository/ClubRepositoryIT.java:417]: this test goes through all the parameters ..... set it to value and check if it works

**Review by marianben — COMMENTED** [review]
please pay attention to my comments and give an answer.
It is also interesting what will happen if the field name will have an empty string just not "null"

File: src/main/java/com/softserve/teachua/repository/ClubRepository.java (hunk 2 above)
Comment (lyzunyktaras) [reply src/main/java/com/softserve/teachua/repository/ClubRepository.java:52]: Here you could see the result of advanced search for name: "American" and "Школа"

image

image

image

File: src/main/java/com/softserve/teachua/dto/search/AdvancedSearchClubProfile.java (hunk 1 above)
Comment (lyzunyktaras) [reply src/main/java/com/softserve/teachua/dto/search/AdvancedSearchClubProfile.java:12]: Unfortunately i forgot about that, will fix it now

File: src/main/java/com/softserve/teachua/repository/ClubRepository.java (hunk 2 above)
Comment (marianben) [reply src/main/java/com/softserve/teachua/repository/ClubRepository.java:52]: Does it also work at the centers?

File: src/test/java/com/softserve/teachua/repository/ClubRepositoryIT.java (hunk 3 above)
Comment (lyzunyktaras) [reply src/test/java/com/softserve/teachua/repository/ClubRepositoryIT.java:417]: Works correctly, thanks for your participation and help

**Review by softservedata — APPROVED** [review]

**Pull request merged**